novice-r-practice.Rmd

```diff
@@ -0,0 +1,375 @@
+
+7. Find the dates of the highest radiation reading at each site:
+
+```{r dates-high-rad}
+cleaned %>%
+  dplyr::filter(quantity == "rad") %>%
+  dplyr::group_by(site_id) %>%
+  dplyr::mutate(max_rad = max(reading)) %>%
+  dplyr::filter(reading == max_rad)
+```
+
+Another way to do it:
+
+```{r dates-high-rad-2}
+cleaned %>%
+  dplyr::filter(quantity == "rad") %>%
+  dplyr::group_by(site_id) %>%
+  dplyr::top_n(1, reading) %>%
+  dplyr::select(site_id, visit_date, reading)
+```
+
+*Explain why this __doesn't__ work.*
+
+```{r error-high-rad, error=TRUE}
+cleaned %>%
+  dplyr::filter(quantity == "rad") %>%
+  dplyr::group_by(site_id) %>%
+  dplyr::summarize(max_rad = max(reading)) %>%
+  dplyr::ungroup() %>%
+  dplyr::filter(reading == max_rad)
+```
+
+8. Normalize radiation against the highest radiation seen per site:
+
+```{r normalize-rad}
+cleaned %>%
+  dplyr::filter(quantity == "rad") %>%
+  dplyr::group_by(site_id) %>%
+  dplyr::mutate(
+    max_rad = max(reading),
+    frac_rad = reading / max_rad) %>%
+  dplyr::select(visit_id, site_id, visit_date, frac_rad)
+```
+
+*Normalize salinity against mean salinity by site.*
+
+9. Find stepwise change in radiation per site by date:
+
+```{r rad-change}
+cleaned %>%
+  dplyr::filter(quantity == "rad") %>%
+  dplyr::group_by(site_id) %>%
+  dplyr::mutate(delta_rad = reading - dplyr::lag(reading)) %>%
+  dplyr::arrange(site_id, visit_date)
+```
+
+*Find length of time between visits by site.*
+
+10. Find sites that experience any stepwise increase in radiation between visits:
+
+```{r rad-increases}
+cleaned %>%
+  dplyr::filter(quantity == "rad") %>%
+  dplyr::group_by(site_id) %>%
+  dplyr::mutate(delta_rad = reading - dplyr::lag(reading)) %>%
+  dplyr::filter(!is.na(delta_rad)) %>%
+  dplyr::summarize(any_increase = any(delta_rad > 0)) %>%
+  dplyr::filter(any_increase)
+```
+
+*Find sites with visits more than one year apart.*
+
+## Creating charts
+
+We will use data on the mass and home range area (HRA) of various species from:
+
+> Tamburello N, Côté IM, Dulvy NK (2015) Data from: Energy and the scaling of animal space use. Dryad Digital Repository.
+> https://doi.org/10.5061/dryad.q5j65
+
+```{r}
+hra <- readr::read_csv(here::here("novice-r-practice", "home-range-database.csv"))
+head(hra)
+```
+
+1. Look at how mass is distributed:
+
+```{r chart-mass}
+ggplot2::ggplot(hra) +
+  ggplot2::geom_histogram(mapping = aes(x = mean.mass.g))
+```
+
+Try again with `log10.mass`:
+
+```{r chart-log-mass}
+ggplot2::ggplot(hra) +
+  ggplot2::geom_histogram(mapping = aes(x = log10.mass))
+```
+
+*Create histograms showing the distribution of home range area using linear and log scales.*
+
+2. Change the visual appearance of a chart:
+
+```{r change-visual}
+ggplot2::ggplot(hra) +
+  ggplot2::geom_histogram(mapping = aes(x = log10.mass), bins = 100) +
+  ggplot2::ggtitle("Frequency of Species Masses") + ggplot2::xlab("Log10 of Mass") + ggplot2::ylab("Number of Species") +
```

Comment:
```suggestion
  ggplot2::ggtitle("Frequency of Species Masses") + 
  ggplot2::xlab("Log10 of Mass") + 
  ggplot2::ylab("Number of Species") +
```

Reply:
Newline after each `+` as per the tidyverse coding style. https://style.tidyverse.org/

Reply:
fixed.